### PR TITLE
Shareable URL with severity, color blindness type and test site url

### LINKED ISSUE #3: Support input of test site via a11y-viewer URL fragment
This would allow you to share a link to the a11y-viewer with a site opened inside it.

Example: `https://voorhoede.github.io/a11y-viewer/#url=https://www.voorhoede.nl`

So update has at the same time iframe source is changed.

Change iframe source to url in hash on viewer load (and hash change?)

### PR DESCRIPTION
**Pull request by reinoute**
Resolves #3 and #18.

This includes updating browser history (using History API) in browsers that support it.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "build": "npm-run-all --parallel build:css build:html build:js",
     "build:html": "ncp ./src/index.html ./dist/index.html && ncp ./src/assets/svg-filters.svg ./dist/svg-filters.svg",
     "build:css": "node scripts/postcss.js",
-    "build:js": "uglifyjs --compress --output ./dist/index.js -- ./src/assets/a11y-viewer.js",
+    "build:js": "ncp ./src/assets/a11y-viewer.js ./dist/index.js",
```
Comment (decrek) [inline package.json]: i would uglify, if you need non uglified js in development. Create a seperate issue for development/production build

File: src/assets/a11y-viewer.css [hunk 2]
```diff
@@ -277,143 +278,115 @@ iframe {
 /* filters */
 
 /* moderate visual impairment combinations */
-.moderate-visual-impairment {
+.moderate {
     filter: blur(var(--moderate-blur));
 }
-
-.moderate-visual-impairment.protanopia {
+.moderate.nored {
```
Comment (decrek) [inline src/assets/a11y-viewer.css]: no-red? applies to all classes below

File: src/assets/a11y-viewer.js [hunk 3]
```diff
@@ -1,58 +1,182 @@
...
+            updateHistory(input.id, value);
+        }
     }
 
-    function emulateVisualImparity(event) {
-        var imparityLevel = event.target.value;
+    function simulateVisualImpairment(value) {
 
-        [].forEach.call(imparitySelector.options, function(option) {
+        arrayFrom(visionInput.options).forEach(function (option) {
```
Comment (decrek) [inline src/assets/a11y-viewer.js]: on initialisation?

File: src/assets/a11y-viewer.js [hunk 4]
```diff
@@ -1,58 +1,182 @@
...
-        iframe.classList.add(imparityLevel);
+        iframe.classList.add(value);
     }
 
-    function emulateColorBlindness(event) {
-        var colorBlindnessLevel = event.target.value;
+    function simulateColorBlindness(value) {
 
-        [].forEach.call(colorBlindnessSelector.options, function(option) {
+        arrayFrom(colorInput.options).forEach(function (option) {
```
Comment (decrek) [inline src/assets/a11y-viewer.js]: on initialisation?

File: src/assets/a11y-viewer.js [hunk 5]
```diff
@@ -1,58 +1,182 @@
...
-    var main = document.getElementById('main');
-    var clickStealerToggler = document.getElementById('click-stealer-toggler');
+
+    var VISION_SELECTOR = '[data-input-vision]';
+    var COLOR_SELECTOR = '[data-input-color]';
+    var KEYBOARD_ONLY_SELECTOR = '[data-input-keyboard-only]';
+    var URL_SELECTOR = '[data-input-url]';
+    var DISABLED_CLASS = 'is-disabled';
+
+    var visionInput = document.querySelector(VISION_SELECTOR);
```
Comment (decrek) [inline src/assets/a11y-viewer.js]: visionSelect/visionSelector

File: src/assets/a11y-viewer.js [hunk 6]
```diff
@@ -1,58 +1,182 @@
...
     }
 
-    function submitHandler(e) {
-        e.preventDefault();
-        var url = form.url.value;
+    function updateTestSite(value) {
+        var parsedUrl = parseTestSiteUrl(value);
+        // replace the iframe element with a new one to prevent a new entry in browser history
+        var parent = iframe.parentNode;
+        var newIframe = document.createElement('iframe');
```
Comment (decrek) [inline src/assets/a11y-viewer.js:128]: `¯\_(ツ)_/¯`

**Comment by decrek** [comment]
hmmm, we can not share urls like this. Because on load you call replacestate based on the input values in the html, these are default. So always voorhoede.nl, no filters applied.

File: package.json (hunk 1 above)
Comment (reinoute) [inline package.json]: Done

File: src/assets/a11y-viewer.css (hunk 2 above)
Comment (reinoute) [inline src/assets/a11y-viewer.css]: done, form / url values are updated as well

File: src/assets/a11y-viewer.js (hunk 4 above)
Comment (reinoute) [inline src/assets/a11y-viewer.js]: done

File: src/assets/a11y-viewer.js (hunk 3 above)
Comment (reinoute) [inline src/assets/a11y-viewer.js]: done

File: src/assets/a11y-viewer.js (hunk 5 above)
Comment (reinoute) [inline src/assets/a11y-viewer.js]: done

**Comment by jbmoelker** [comment]
Without going into this issue/PR itself may I suggest we use [squash and merge](https://twitter.com/toddmotto/status/763263961101991936) to keep our history clean (a `fix()` isn't:

squash-and-merge

**Pull request merged**